frontend/src/lib/components/Select/select.tsx

```diff
@@ -48,16 +48,24 @@ export const Select = withDefaults<SelectProps>()(defaultProps, (props) => {
     const [startIndex, setStartIndex] = React.useState<number>(0);
     const [lastShiftIndex, setLastShiftIndex] = React.useState<number>(-1);
     const [currentIndex, setCurrentIndex] = React.useState<number>(0);
+    const [options, setOptions] = React.useState<SelectOption[]>(props.options);
+    const [prevOptions, setPrevOptions] = React.useState<SelectOption[]>(props.options);
     const [prevFilteredOptions, setPrevFilteredOptions] = React.useState<SelectOption[]>([]);
 
     const ref = React.useRef<HTMLDivElement>(null);
     const noOptionsText = props.placeholder ?? "No options";
+
+    if (!isEqual(props.options, prevOptions)) {
+        setPrevOptions(props.options);
```

Comment:
Is it need for `prevOptions`, as both `options` and `prevOptions` states are set simultaneously?

NB: on line 231 below, the `props.options` are used in the returned component render code. Should it be using `options` state instead?

Reply:
Good observation, seems like a relict from older days ;). Line 231 would not matter, but I can change it to have it more concise.